File: src/scala/func.scala
object LVFunc {

  case class MeetupGuest(name: String, age: Int, techno: List[String])

  val scala = "scala"
  val java = "java"
  val php = "php"
  val rust = "rust"

  val guests = List(
    MeetupGuest("Rocky", 38, List(scala, java)),
    MeetupGuest("Apollo", 42, List(rust, php, java)),
    MeetupGuest("Adrienne", 34, List(php, rust)),
    MeetupGuest("Drago", 26, List(rust)),
    MeetupGuest("Mary Anne", 32, List(java, php))
  )

  def printAge(guest: MeetupGuest): String = s"I'm ${guest.age}"

  def printName(guest: MeetupGuest): String = s"Hello ${guest.name} !"

  def printTechno(guest: MeetupGuest, join: String): String = s"I craft with ${guest.techno.mkString(join)}"

  def print(guests: List[MeetupGuest], format: MeetupGuest => String): Unit =
    println(
      guests
        .foldRight[List[String]](Nil)((c, r) => format(c) :: r)
        .mkString("\n")
    )

  def main(args: Array[String]): Unit = {
    print(guests, printAge)
    print(guests, printName)
    print(guests, printTechno(_: MeetupGuest, " - "))
  }

  def ln = println("")

}

File: src/php/func.php
<?php

class MeetupGuest
{
    function __construct($name, $age, array $techno)
    {
        $this->name = $name;
        $this->age = $age;
        $this->techno = $techno;
    }

    function printInfo($printType)
    {
        switch ($printType) {
            case 'printAge':
                echo $this->age;
                break;
            case 'printName':
                echo $this->name;
                break;
            case 'printTechno':
                echo implode('', $this->techno);
                break;
            default:
                echo print('%s is invalid');
        }
    }
}

$scala = "scala";
$java = "java";
$php = "php";
$rust = "rust";

$guests = [
    new MeetupGuest("Rocky", 38, [$scala, $java]),
    new MeetupGuest("Apollo", 42, [$rust, $php, $java]),
    new MeetupGuest("Adrienne", 34, [$php, $rust]),
    new MeetupGuest("Drago", 26, [$rust]),
    new MeetupGuest("Mary Anne", 32, [$java, $php])
];

$printActions = ['printAge', 'printName', 'printTechno'];

foreach ($printActions as $printAction) {
    foreach ($guests as $guest) {
        $guest->printInfo($printAction);
        ln();
    }
    ln();
}

function ln()
{
    echo "\n";
}
